internal/git/git.go

```diff
@@ -265,12 +266,37 @@ func (g *GitOperations) IsPullRequestRequired(ctx context.Context, environmentNe
 	}
 	logger.V(4).Info("Got diff", "diff", stdout)
 
-	// Check if the diff contains any YAML files if so we expect a manifest to have changed
-	// TODO: This is temporary check we should add some path globbing support to the specs
-	for _, file := range strings.Split(stdout, "\n") {
-		if strings.HasSuffix(file, ".yaml") || strings.HasSuffix(file, ".yml") {
-			logger.V(4).Info("YAML file changed", "file", file)
-			return true, nil
+	openPullRequest, err := OpenPullRequestFilter(ctx, filter, strings.Split(stdout, "\n"))
+	if err != nil {
+		return false, fmt.Errorf("failed to filter if we should open pull request: %w", err)
+	}
+
+	return openPullRequest, nil
+}
+
+func OpenPullRequestFilter(ctx context.Context, filter *v1alpha1.OpenPullerRequestFilter, diffOutput []string) (bool, error) {
+	logger := log.FromContext(ctx)
+	if filter == nil {
+		logger.V(1).Info("No filter provided, defaulting to checking for yaml files")
+		for _, file := range diffOutput {
+			if strings.HasSuffix(file, ".yaml") || strings.HasSuffix(file, ".yml") {
+				logger.V(4).Info("YAML file changed", "file", file)
+				return true, nil
+			}
+		}
+		return false, nil
+	}
+
+	for _, file := range diffOutput {
+		for _, pathPattern := range filter.Paths {
+			matched, err := filepath.Match(pathPattern, file)
```

Comment:
We should probably think about how we can mitigate directory traversal risks when applying glob patterns. Since we return a boolean, I think the only risk is path enumeration.